packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -585,6 +516,96 @@ bool validateJavaGradle(Logger logger,
   return false;
 }
 
+// Returns compatibility information for the valid range of Gradle versions for
+// the specified Java version.
```

Comment:
Can you make this `///` for a public dart doc and also add a note about when it returns null and what that means for the caller?